torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -811,17 +817,23 @@ def benchmark_one_config(config):
         best_config = self.coordesc_tuner.autotune(
             benchmark_one_config, launcher.config, None
         )
-        time_taken_ns = time.time_ns() - start_time
+        coordesc_time_taken_ns = time.time_ns() - start_time
         best_config.found_by_coordesc = True
 
         if self.save_cache_hook:
-            self.save_cache_hook(best_config, time_taken_ns, found_by_coordesc=True)
+            self.save_cache_hook(
+                best_config,
+                self.autotune_time_taken_ns + coordesc_time_taken_ns,
+                found_by_coordesc=True,
+            )
         return config2launcher.get(best_config)
 
     def run(self, *args, grid, stream, **kwargs):
         if len(self.launchers) != 1:
             if len(self.launchers) == 0:
+                start_time = time.time_ns()
```

Comment:
maybe someday we should wrap the NS/Millisecond etc in classes.. a la [typed units of measure](https://varkor.github.io/blog/2018/07/30/types-for-units-of-measure.html). feel like it would be so easy to miss a ns/ms mismatch

Reply:
for now, i resort to suffixing all my units but i'd love to use a strongly typed language to avoid these suffix games :)

Reply:
typed units of measure only exists in F# anyway.

Reply:
^^  #1 python defender (although someone else can defend mypy..)

Reply:
I guess in a compiled language we could have built zero cost abstractions, similar to rust snippet on your link above, but i assume in python, there would be a cost to wrapping it with classes etc

Reply:
> typed units of measure only exists in F# anyway. and python is strongly typed - it's just dynamically typed. theres nothing preventing us from doing this.

Not just F# - lots of languages over the years have had that (I worked on one MANY years ago). C++ even supports it: https://www.boost.org/doc/libs/1_82_0/doc/html/boost_units.html

Reply:
+1 on what @oulgen said about the cost of wrapping in a class (although if done right this would probably not matter for our use case); we might also be able to do some tricky `if TYPE_CHECKING` stuff to get the types checked without changing the actual runtime, but that would probably be messy